nipype/algorithms/confounds.py

```diff
@@ -1201,20 +1281,56 @@ def compute_noise_components(imgseries, mask_images, num_components,
         # "The covariance matrix C = MMT was constructed and decomposed into its
         # principal components using a singular value decomposition."
         try:
-            u, _, _ = fallback_svd(M, full_matrices=False)
+            u, s, _ = fallback_svd(M, full_matrices=False)
         except np.linalg.LinAlgError:
-            if self.inputs.failure_mode == 'error':
+            if failure_mode == 'error':
                 raise
-            u = np.ones((M.shape[0], num_components), dtype=np.float32) * np.nan
+            if components_criterion >= 1:
+                u = np.empty((M.shape[0], components_criterion),
+                             dtype=np.float32) * np.nan
```

Comment:
To accommodate above suggestion:

```suggestion
```

Also `s` is not defined in this case.

Reply:
Out-of-order. ["Above suggestion"](https://github.com/nipy/nipype/pull/2878#discussion_r259085346).